### PR TITLE
DOC: Clarify index.html

### PR DESCRIPTION
**Pull request by bjnath**
Explicit link to current stable docs, explicit links to its pdf.
Clarify that devdocs is unstable. Update copyright line.

<details><summary>New page</summary>

image

</details>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -42,41 +42,40 @@
...
-<p>Other links:</p>
 <ul class="simple">
-  <li>
-    <a class="biglink" href="/neps">NumPy Enhancement Proposals</a>
-       (which include the NumPy Roadmap and detailed plans for major new features).
-  </li>
+  <p><li><a class="biglink" href="https://numpy.org/doc/stable/">Web</a></li></p>
+  <p><li><a class="biglink" href="https://numpy.org/doc/stable/numpy-ref.pdf">Reference Guide PDF</a></li></p>
+  <p><li><a class="biglink" href="https://numpy.org/doc/stable/numpy-user.pdf">User Guide PDF</a></li></p>
+  <p><li><a class="biglink" href="https://numpy.org/devdocs/">Latest (unstable) documentation</a></li></p>
```
Comment (rossbar) [inline index.html:74]:
```suggestion
  <p><li><a class="biglink" href="https://numpy.org/devdocs/">Latest (development) documentation</a></li></p>
```

**Comment by mattip** [comment]
Thanks @bjnath

**Comment by mattip** [comment]
Just a note: this page is not generated, it is manually updated when we release a new version, brief instructions are [here](https://numpy.org/devdocs/dev/releasing.html?highlight=how#build-and-archive-documentation)

**Comment by bjnath** [comment]
+1 for unstable -> development

Thanks.

**Pull request merged**